src/main/java/com/tterrag/registrate/providers/RegistrateRecipeProvider.java

```diff
@@ -101,81 +93,81 @@ public String safeName(ItemLike registryEntry) {
     public static final int DEFAULT_BLAST_TIME = DEFAULT_SMELT_TIME / 2;
     public static final int DEFAULT_SMOKE_TIME = DEFAULT_BLAST_TIME;
     public static final int DEFAULT_CAMPFIRE_TIME = DEFAULT_SMELT_TIME * 3;
-    
+
     private static final String SMELTING_NAME = "smelting";
     @SuppressWarnings("null")
     private static final ImmutableMap<SimpleCookingSerializer<?>, String> COOKING_TYPE_NAMES = ImmutableMap.<SimpleCookingSerializer<?>, String>builder()
-            .put(RecipeSerializer.SMELTING_RECIPE, SMELTING_NAME)
-            .put(RecipeSerializer.BLASTING_RECIPE, "blasting")
-            .put(RecipeSerializer.SMOKING_RECIPE, "smoking")
-            .put(RecipeSerializer.CAMPFIRE_COOKING_RECIPE, "campfire")
+            .put((SimpleCookingSerializer<?>) RecipeSerializer.SMELTING_RECIPE, SMELTING_NAME)
+            .put((SimpleCookingSerializer<?>) RecipeSerializer.BLASTING_RECIPE, "blasting")
+            .put((SimpleCookingSerializer<?>) RecipeSerializer.SMOKING_RECIPE, "smoking")
+            .put((SimpleCookingSerializer<?>) RecipeSerializer.CAMPFIRE_COOKING_RECIPE, "campfire")
```

Comment:
This shouldn't be necessary? It's an implicit type widening.

Reply:
Intelij will not compile without the cast

```java
G:\Development\Java\Minecraft\Registrate\src\main\java\com\tterrag\registrate\providers\RegistrateRecipeProvider.java:103: error: incompatible types: RecipeSerializer<SmeltingRecipe> cannot be converted to SimpleCookingSerializer<?>
            .put(RecipeSerializer.SMELTING_RECIPE, SMELTING_NAME)
```

feel like this is because it is stored as `RecipeSerializer` now rather than `SimpleCookSerializer`

> 1.19.3
`RecipeSerializer<SmeltingRecipe> SMELTING_RECIPE = register("smelting", new SimpleCookingSerializer<>(SmeltingRecipe::new, 200));`

> 1.19.2
`SimpleCookingSerializer<SmeltingRecipe> SMELTING_RECIPE = register("smelting", new SimpleCookingSerializer<>(SmeltingRecipe::new, 200));`

Reply:
Oh, in that case, is the subtype being used in the map necessary? Do we use any methods specific to SimpleCookingSerializer?

Reply:
The methods the map is used in lead to `SimpleCookingRecipeBuilder#generic` which takes in a `RecipeSerializer<? extends AbstractCookingRecipe> p_250921_` parameter.

So yeah I think we are going to need a cast somewhere to the cooking recipe type